src/nash_equilibrium.jl

```diff
@@ -0,0 +1,23 @@
+"""
+Finds all pure strategy Nash equilibrium for a normal form
+game. It returns an empty array if there are no nash
+equilibrium
+"""
+function pure_strategy_NE(nfg::NormalFormGame)
+    # Get number of players and their actions
+    np = num_players(nfg)
+    na = nfg.nums_actions
+
+    # Holder for all NE
+    ne = Array(ActionProfile, 0)
+
+    # For each action profile check whether it is NE
+    as = CartesianRange(na)
+    for a in as
+        _a = a.I::NTuple{np, Int}
```

Comment:
Having to insert type checks is never a great thing because (I believe) it incurs minor runtime cost.

It might be better to add a method `is_nash{N}(::NormalFormGame{N}, ::CartesianIndex{N})` and then we can skip the `_a` bit all together